qiskit/pulse/configuration.py

```diff
@@ -14,11 +14,28 @@
 Configurations for pulse experiments.
 """
 from typing import Dict, Union, Tuple, Optional
+import numpy as np
 
 from .channels import PulseChannel, DriveChannel, MeasureChannel
 from .exceptions import PulseError
 
 
+def _assert_nested_dict_equal(a, b):
+    for key in a:
+        if key in b:
+            if isinstance(a[key], dict):
+                _assert_nested_dict_equal(a[key], b[key])
```

Comment:
We need to propagate up the case where they are not equal:

```suggestion
                if not _assert_nested_dict_equal(a[key], b[key]):
                    return False
```

Reply:
Thank you very much for pointing out my error! 🙇